src/redux/actions/samples/updateSample.js

```diff
@@ -3,34 +3,84 @@ import _ from 'lodash';
 
 import endUserMessages from 'utils/endUserMessages';
 
+import {
+  SAMPLES_UPDATE, SAMPLES_SAVING, SAMPLES_SAVED,
+} from 'redux/actionTypes/samples';
+import saveSamples from 'redux/actions/samples/saveSamples';
+
 import mergeObjectWithArrays from 'utils/mergeObjectWithArrays';
 import handleError from 'utils/http/handleError';
-import {
-  SAMPLES_UPDATE,
-} from '../../actionTypes/samples';
-import saveSamples from './saveSamples';
+import fetchAPI from 'utils/http/fetchAPI';
+
+import config from 'config';
+import { api } from 'utils/constants';
+
+const sendApiV2Request = async (experimentId, sampleId, diff, dispatch) => {
+  const url = `/v2/experiments/${experimentId}/samples/${sampleId}`;
+  const body = diff;
+
+  dispatch({
+    type: SAMPLES_SAVING,
+    payload: {
+      message: endUserMessages.SAVING_SAMPLE,
+    },
+  });
+
+  await fetchAPI(
+    url,
+    {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(body),
+    },
+  );
+
+  dispatch({
+    type: SAMPLES_SAVED,
+  });
+
+  dispatch({
+    type: SAMPLES_UPDATE,
+    payload: {
+      sampleId,
+      sample: diff,
+    },
+  });
+};
```

Comment:
if the version of the api is v2, then the `SAMPLES_UPDATE` action will be dispatched twice: once here and once in lines 77-83